### DISCUSSION TITLE
How can I customize website page titles?

### DISCUSSION THREAD

**Discussion by seeM** [quarto-dev/quarto-cli#5260] [Q&A] [answered]
IIUC posts rendered as website pages have titles always in the format: `<website title> - <post title>`. I would like the option to customize this.

E.g. for my personal website, social preview cards render like so:

<img width="610" alt="image" src="https://user-images.githubusercontent.com/559360/233345684-da767bd9-c4d0-43d0-97a6-277018312cf8.png">

But I'd prefer if the `Wasim Lorgat - ` part weren't there, since the website URL already has my name 😄

**Comment by cderv** [comment]
You are referring to Twitter Card or Open Graph based social card, right ? 

Did you already customized the values per our doc ? https://quarto.org/docs/reference/projects/websites.html#social

By default the title of the page will be used, but you can overwrite using those configuration.

**Comment by seeM** [comment]
I’m referring to the same title that seems to be used for the HTML web page and the Twitter and open graph cards. I could override each of those for every single post but I’d prefer if it were possible to change the string template once e.g in my `_quarto.yml`.

**Comment by cderv** [comment]
I think what you are looking for is the [metadata](https://quarto.org/docs/reference/formats/html.html#metadata) configuration. 

By default 
* We build {website.tite} - {title}
* You can change that to `{title-prefix} - {title}` by providing a `title-prefix` value
* You can completely overwrite this by providing a `pagetitle` configuration. If set, then this will be used. 

So you could try set a `pagetitle` in your post and this should change what you see in HTML tab in browser, and in Social card. 

You cannot change the string template directly to something based on other meta for now

**Comment by mcanouil** [comment, marked as answer]
Related to #1736 in some aspects.

In `_quarto.yml`, you can add `pagetitle: "{{< meta title >}}"` to overwrite `pagetitle` and remove the website title prefix.

@cderv about `title-prefix` there is currently an issue, see #5196.
PS: Maybe worth to transfer this issue to "Discussions Q&A".

**Comment by cderv** [comment]
Good catch for the issue. It seems it is not working as I expected or as it should... 
We'll follow that in #5196. 

@mcanouil workaround is a good one for your need @seeM, I believe it answers your problem, isn't it ?

**Comment by ghost** [comment]
@cderv @mcanouil Should we close or move this issue?

**Comment by cderv** [comment]
Let's move it to question as suggested by @mcanouil - there is no real issue attached to it.

**Comment by seeM** [comment]
> In _quarto.yml, you can add pagetitle: "{{< meta title >}}" to overwrite pagetitle and remove the website title prefix.

Thanks, this works perfectly! 😄

**Reply by mcanouil** [reply]
Note that this is really a workaround, since once #5196 is fixed you should be able to remove prefix using the `title-prefix` option.

**Reply by jolars** [reply]
Setting `title-prefix` only seems to work (now that #5196 is closed) if you want to modify the title-prefix but not remove it altogether. At least setting it as an empty string doesn't do anything for me.

**Reply by amitness** [reply]
@mcanouil The workaround seems to have broken with the latest quarto version.

This option in _quarto.yml doesn't work anymore
```yaml
metadata:
  pagetitle: "{{< meta title >}}"
```

I tried setting title-prefix to "" in _quarto.yml and that didn't work as well.
```yaml
metadata:
    title-prefix: ""
```

Instead, I had to create a _metadata.yml file in my posts folder and set the title-prefix there. That fixed it for now.
```yaml
title-prefix: ""
```

Not sure, if it's a bug due to recent release or expected behavior.

**Reply by cderv** [reply]
@amitness can you open an issue with a reproducible example of what you are doing, and sharing the Quarto version used. 

We release quarto 1.5 earlier this week so be sure to update. Thanks !

**Reply by xxuu0086** [reply]
@amitness Thank you!   It works in _quarto.yml without "metadata:"
```
format:
  html:
    title-prefix: ""
```
